src/crypto/crypto_util.cc

```diff
@@ -206,21 +195,14 @@ void InitCryptoOnce() {
   sk_SSL_COMP_zero(SSL_COMP_get_compression_methods());
 
 #ifndef OPENSSL_NO_ENGINE
-  ERR_load_ENGINE_strings();
-  ENGINE_load_builtin_engines();
+  ncrypto::EnginePointer::initEnginesOnce();
 #endif  // !OPENSSL_NO_ENGINE
 }
 
 void GetFipsCrypto(const FunctionCallbackInfo<Value>& args) {
   Mutex::ScopedLock lock(per_process::cli_options_mutex);
   Mutex::ScopedLock fips_lock(fips_mutex);
-
-#if OPENSSL_VERSION_MAJOR >= 3
-  args.GetReturnValue().Set(EVP_default_properties_is_fips_enabled(nullptr) ?
-      1 : 0);
-#else
-  args.GetReturnValue().Set(FIPS_mode() ? 1 : 0);
-#endif
+  args.GetReturnValue().Set(ncrypto::isFipsEnabled() ? 1 : 0);
```

Comment:
You can cast this to uint32 and get the same thing without a branch

Reply:
The branch will be optimized away by the compiler, so there's not much of an upside of a `static_cast`.

Reply:
For education purposes: how can I validate?

Reply:
Maybe @lemire?

Reply:
According to https://godbolt.org/ using the static_cast is slightly more optimized with two less operations, but they are both pretty close. https://godbolt.org/z/6561MW3jj

Reply:
And that's with optimizations disabled. With optimizations, they tend to result in the same code: https://godbolt.org/z/v9oqnvWoK

Reply:
doh... that's right, I forgot to turn on the optimizations

Reply:
Another educational question: Why do we have a mutex lock in this function, and why shouldn't we do it inside `crypto::isFipsEnabled`?

Reply:
I don't want to put any mutex logic into ncrypto so definitely do not want to move it into the new `ncrypto::isFipsEnabled()` method. As for why the mutex is here at all, it's possible that this could end up being called from any number of worker threads. Really needs to be limited to one at a time but there are possibly some more improvements here. I don't think this actually needs the `cli_options_mutex` but would need to verify